tests/utils/output_verification.hpp

```diff
@@ -141,36 +143,37 @@ int vector_verification(
 	const constexpr T one = static_cast< T >( 1 );
 
 	// open verification file
-	FILE * const in = fopen( truth_filename, "r" );
+	std::ifstream in;
+	in.open( truth_filename);
 
-	if( in == nullptr ) {
-		std::cerr << "Could not open the file \"" << truth_filename << "\"."
+	if( !in.is_open() ) {
+		std::stringstream error;
+		error << "Could not open the file \"" << truth_filename << "\"."
 			<< std::endl;
-		return 10;
+		throw std::runtime_error(error.str());
 	}
 
 	// read the truth output vector from the input verification file
 	const size_t n = output_vector.size();
 	T * const truth = new T[ n ];
 	if( truth == nullptr ) {
 		std::cerr << "Could not allocate necessary buffer" << std::endl;
-		return 20;
+		throw std::bad_alloc();
 	}
 
 	for( size_t i = 0; i < n; i++ ) {
 		const int rc = data_fscanf( in, truth + i );
 		if( rc != 0 ) {
-			std::cerr << "The verification file looks incomplete. " << "Line i = " << i
+			std::stringstream error;
+			error << "The verification file looks incomplete. " << "Line i = " << i
 				<< ", data = " << truth[ i ] << ", rc = " << rc << std::endl;
 			delete [] truth;
```

Comment:
Canonically, should this `delete []` not go just after a try-block?

Reply:
Not sure what you're implying since there's no exception thrown around here

Reply:
Just to clarify: if all errors were to be exceptions, then this block should throw. We could then throw before deleting truth, and handle memory reclaiming by catching all errors after allocating truth, deleting truth, and then rethrow. But as per the earlier remark, not everything should have to be an exception so the current pattern is fine=)